client/src/test/java/org/apache/rocketmq/client/impl/factory/MQClientInstanceTest.java

```diff
@@ -74,6 +87,34 @@ public void testTopicRouteData2TopicPublishInfo() {
         assertThat(topicPublishInfo.getMessageQueueList().size()).isEqualTo(4);
     }
 
+    @Test
+    public void testFindBrokerAddressInSubscribe() {
+        // dledger normal case
+        String brokerName = "BrokerA";
+        HashMap<Long, String> addrMap = new HashMap<Long, String>();
+        addrMap.put(0L, "127.0.0.1:10911");
+        addrMap.put(1L, "127.0.0.1:10912");
+        addrMap.put(2L, "127.0.0.1:10913");
+        brokerAddrTable.put(brokerName, addrMap);
+        long brokerId = 1;
+        FindBrokerResult brokerResult = mqClientInstance.findBrokerAddressInSubscribe(brokerName, brokerId, false);
+        assertThat(brokerResult).isNotNull();
+        assertThat(brokerResult.getBrokerAddr()).isEqualTo("127.0.0.1:10912");
+        assertThat(brokerResult.isSlave()).isTrue();
+
+        // dledger case, when node n0 was voted as the leader
+        brokerName = "BrokerB";
+        HashMap<Long, String> addrMapNew = new HashMap<Long, String>();
+        addrMapNew.put(0L, "127.0.0.1:10911");
+        addrMapNew.put(2L, "127.0.0.1:10912");
```

Comment:
Is there any difference in your test case?(0,1,2 vs 0,2,3)

Reply:
(0,1,2) is the normal case, which is added to check my code and also for the future changes from others. IMO, it can improve the coverage of UT.
(0,2,3) is the "bugfix" case, which is added to check if my code fix the bug.

Reply:
Please go ahead, I would like to follow your pr :-)